### ISSUE TITLE
Supporting custom env vars for components

### ISSUE THREAD

**Issue by mousavian** [banzaicloud/istio-operator#294]
**Is your feature request related to a problem? Please describe.**
Some components of istio come with some tuning flags that can be set through ENV variables. 
(eg [pilot](https://istio.io/docs/reference/commands/pilot-discovery/#envvars), [galley](https://istio.io/docs/reference/commands/galley/#envvars), etc.) as well as some env vars that affect [Golang runtime](https://dave.cheney.net/2015/11/29/a-whirlwind-tour-of-gos-runtime-environment-variables).
In my case, I need to provide `GOMAXPROCS` for all components, specifically pilot/discovery container as we see a lots of CPU throttling happening there.

Similar to #216  but in more generic way.

**Describe the solution you'd like to see**
I would like to be able to set env variables as key value in my kustomization for each component under `Kind: Istio`. Similar to what [istio/installer using Helm](https://github.com/istio/istio/blob/e751268/install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml#L104-L109) is providing.

**Describe alternatives you've considered**

**Additional context**
Manually adding `GOMAXPROCS` to the right value in pilot deployment helped us to reduce cpu throttling and memory consumption at the same time.

<img width="558" alt="Screen Shot 2019-09-11 at 13 07 10" src="https://user-images.githubusercontent.com/317322/64665226-2061b280-d495-11e9-9b31-c036a1188a2e.png">

```
rate(container_cpu_cfs_throttled_seconds_total{namespace="$namespace",container_name="$container"}[5m])
```

**Comment by Laci21** [comment]
Hi @mousavian,

This feature would make sense for sure.

We will do our best to implement this feature in the next 2-3 weeks, but in around two weeks we will have a [Backyards](https://banzaicloud.com/blog/istio-the-easy-way/) release, until then we have limited capacity, so we cannot promise anything. If you need the feature sooner and you are up for it, we are more than happy to accept PRs as you already know. Otherwise, we will try to resolve this as soon as we can.

**Comment by BoOgatti** [comment]
Hi,
In my case it will be useful for enabling HTTP/1.0 (PILOT_HTTP10: 1). Now I have to doing it manually

**Comment by Laci21** [comment]
Hi @mousavian and @BoOgatti,

Sorry for the unusually long delay with this feature, but finally it was implemented in #374, courtesy of @waynz0r!

You can use the new `additionalEnvVars` field under each component to set any custom environment variables for the Istio control plane components.

This feature will land in the next operator release (which will be on Monday most probably).

**Comment by BoOgatti** [comment]
Thx guys! It works and helpfull. In first time It was slightly not obvious how to implementing this new feature and what kind of array it must be ))

Example, just for history:
```
spec: 
 pilot:
    additionalEnvVars:
      - name: PILOT_HTTP10
        value: "1"
```
